dali/pipeline/executor/executor.h

```diff
@@ -142,6 +145,16 @@ class DLL_PUBLIC Executor : public ExecutorBase, public QueuePolicy {
 
   DISABLE_COPY_MOVE_ASSIGN(Executor);
 
+  /**
+   * Returns the checkpoint for the given iteration ID.
+  */
+  DLL_PUBLIC Checkpoint& GetCurrentCheckpoint(size_t iteration_id);
```

Comment:
I think we should not expose the iteration_id in the public API, should we? We can even have GetCurrentCheckpoint in public API and GetCheckpoint(int iteration_id) in the protected one. Similarily as ShareOutputs gets the trace based on the internally kept output_idx.